src/finally.js

```diff
@@ -103,9 +104,27 @@ Promise.prototype["finally"] = function (handler) {
                              finallyHandler);
 };
 
+
 Promise.prototype.tap = function (handler) {
     return this._passThrough(handler, TAP_TYPE, finallyHandler);
 };
 
+Promise.prototype.tapError = function (handlerOrPredicate) {
+    if(arguments.length === 1) {
+        return this._passThrough(handlerOrPredicate,
+                                 TAP_TYPE,
+                                 undefined,
+                                 finallyHandler);
+    } else {
+        var predicate = arguments[0];
+        var handler = arguments[1];
+        return this._passThrough(handler,
+                                 TAP_TYPE,
+                                 undefined,
+                                 catchFilter(predicate, handler, this));
```

Comment:
wow, I spent like 20 minutes thinking what I'm not understanding in catchFilter and it ended up being a mistype -_-